.cirrus.yml

```diff
@@ -30,16 +32,13 @@ debian-11_task:
     cpu: 1
   fix-pip_script:
     # see https://github.com/pypa/pip/pull/11623
-    - python3 -m pip install --upgrade "pip @ git+https://github.com/pypa/pip.git@a057c9bc99ad7eab2e3b1539e79eee7d9a923f4f"
+    - python3 -m pip install --upgrade "pip>=23.0"
   << : *test
 
 debian-unstable_task:
   container:
     dockerfile: ci/debian-unstable.docker
     cpu: 1
-  fix-pip_script:
-    # see https://github.com/pypa/pip/pull/11623
-    - python3 -m pip install --upgrade "pip @ git+https://github.com/pypa/pip.git@a057c9bc99ad7eab2e3b1539e79eee7d9a923f4f"
```

Comment:
The fact that this fixed itself is very surprising. The only explanation I have is that Debian started distributing a pip with a downstream fix for the bug. I'll need to take a closer look.

Reply:
Yup, that is it. From the `python3-pip` Debian package changelog:
```
python-pip (22.3.1+dfsg-2) unstable; urgency=medium

  * Patch: Use the default sysconfig scheme, in isolated environments.
    (Closes: #1019293)

 -- Stefano Rivera <stefanor@debian.org>  Fri, 20 Jan 2023 15:26:49 -0400
```